### PR TITLE
Promote return chunks of results for list calls to conformance

### LINKED ISSUE #79116: Promote PodTemplate Test to Conformance
Using the list of tested stable endpoints referenced in https://github.com/kubernetes/kubernetes/issues/78748, we can pick out the endpoints which are part of the PodTemplate kind:

[createCoreV1NamespacedPodTemplate](https://apisnoop.cncf.io/?zoomed=operationId-stable-core-createCoreV1NamespacedPodTemplate&showUntested=false&showConformanceTested=false)

This is a stable core endpoint which has existing testing, potentially making for low-hanging fruit to upgrade to Conformance.

/area test
/area conformance
/sig testing

### PR DESCRIPTION
**Pull request by BobyMCbobs**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Promotes the following E2E test to Conformance:  
`test/e2e/apimachinery/chunking.go: "should return chunks of results for list calls"`

**Which issue(s) this PR fixes**:
Fixes #79116

**Special notes for your reviewer**:
Part of Umbrella Issue #78748

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```
/area conformance
/area test
@kubernetes/sig-apimachinery-pr-reviews
@kubernetes/sig-architecture-pr-reviews
@kubernetes/cncf-conformance-wg

### CODE REVIEW

File: test/e2e/apimachinery/chunking.go [hunk 1]
```diff
@@ -75,7 +75,12 @@ var _ = SIGDescribe("Servers with support for API chunking", func() {
 		})
 	})
 
-	ginkgo.It("should return chunks of results for list calls", func() {
+	/*
+		Release : v1.15
```
Comment (timothysc) [inline test/e2e/apimachinery/chunking.go]: It's 1.16

File: test/e2e/apimachinery/chunking.go [hunk 2]
```diff
@@ -75,7 +75,12 @@ var _ = SIGDescribe("Servers with support for API chunking", func() {
...
+		Testname: Return chunks of results for large numbers of objects
+		Description: List a large number of objects and verify the first three chunks
+	*/
+	framework.ConformanceIt("should return chunks of results for list calls", func() {
```
Comment (timothysc) [inline test/e2e/apimachinery/chunking.go]: This description needs a lot of work imo.

Comment (timothysc) [reply test/e2e/apimachinery/chunking.go]: So in looking through the test it's only exercising pod templates and if were going to evaluate behavior I'd argue this test needs to a lot more work.

**Review by timothysc — CHANGES_REQUESTED** [review]
/hold 
/assign @timothysc

File: test/e2e/apimachinery/chunking.go (hunk 2 above)
Comment (hh) [reply test/e2e/apimachinery/chunking.go]: @bobymcbobs can you update with the RFC2119 style description?

@timothysc We need to exercise something to list in order to return chunks, pod templates happens to be what is used here. We could shift to listing something else if something might fit better. It does happen to use an endpoint that no current conformance test hits, hence the prioritization as a possible quick promotion.

Comment (timothysc) [reply test/e2e/apimachinery/chunking.go]: So you could use the dynamic client to hit chunk across multiple resource kinds.  Just promoting the existing test that is poorly named is not sufficient.  1. Make a robust test that exercises a behavior, 2. promote.  I'd argue we this test needs #1 first.

Comment (hh) [reply test/e2e/apimachinery/chunking.go]: The idea of #78748 was to promote existing tests, add metadata and updating descriptions when those tests already hit endpoints that increase coverage.

If this test isn't robust enough, let's go ahead and close it.

I'd rather prioritize writing new tests based on stable endpoints hit by core kubernetes components:
https://github.com/kubernetes/kubernetes/issues/78749.

Comment (timothysc) [reply test/e2e/apimachinery/chunking.go]: We are generally not interested in increasing raw endpoint coverage indirectly, or through poorly written/phrased tests.  We want to be clear and that we are exercising expected behavior that we want to be consistent across providers in a deliberate and well thought out manner. 

/cc @spiffxp

Comment (devaips) [reply test/e2e/apimachinery/chunking.go]: @timothysc can you clarify whether you believe this test is not a good test for the behavior it is setting out to exercise, or if you are saying that it only indirectly covers the endpoint in question (`createCoreV1NamespacedPodTemplate`) hence it should not be considered a good test for *that specific endpoint*. If it is the former, then we will close this PR promptly. If it's the latter, then we could continue to work on promoting this test. Apologies if I'm misunderstanding the comments you've already made, but this is the question which still remains for me.

Comment (spiffxp) [reply test/e2e/apimachinery/chunking.go]: In looking at the issue that motivated opening this PR, if our goal is to cover `createCoreV1NamespacedPodTemplate` this isn't the test to do it.  It _happens_ to do it, but a more straightforward CRUD test would be more appropriate.

**Comment by spiffxp** [comment]
/close
Outcomes of this:
- if we want to exercise the behavior that is "api should chunk list calls" we need this specific test to be rewritten (or split into multiple test cases) for a variety of resources, not just `PodTemplate`
- if we want to exercise PodTemplate creation, we should do so directly in tests that verify pod template creation and resulting behaviors

**Pull request closed without merging**